<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Proteomics.data.analysis">
<title>Label-free Quantification Proteomics • Proteomics.data.analysis</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Label-free Quantification Proteomics">
<meta property="og:description" content="Proteomics.data.analysis">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Proteomics.data.analysis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/LFQ.html">Label-free Quantification Proteomics</a>
    <a class="dropdown-item" href="../articles/SILAC.html">Processing and QC of SILAC data</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="LFQ_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Label-free Quantification Proteomics</h1>
                        <h4 data-toc-skip class="author">Tom Smith</h4>
            
            <h4 data-toc-skip class="date">2022-04-19</h4>
      
      
      <div class="d-none name"><code>LFQ.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://lgatto.github.io/MSnbase" class="external-link">MSnbase</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; Loading required package: mzR</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid, I, unname</span>
<span class="co">#&gt; Loading required package: ProtGenerics</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'ProtGenerics'</span>
<span class="co">#&gt; The following object is masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     smooth</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; This is MSnbase version 2.20.4 </span>
<span class="co">#&gt;   Visit https://lgatto.github.io/MSnbase/ to get started.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'MSnbase'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     trimws</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/StoreyLab/biobroom" class="external-link">biobroom</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: broom</span>
<span class="co">#&gt; Registered S3 methods overwritten by 'biobroom':</span>
<span class="co">#&gt;   method      from </span>
<span class="co">#&gt;   glance.list broom</span>
<span class="co">#&gt;   tidy.list   broom</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">camprotR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mrctoxbioinformatics.github.io/Proteomics_data_analysis/">Proteomics.data.analysis</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following object is masked from 'package:MSnbase':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span>
<span class="co">#&gt; The following objects are masked from 'package:S4Vectors':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     first, intersect, rename, setdiff, setequal, union</span>
<span class="co">#&gt; The following object is masked from 'package:Biobase':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span>
<span class="co">#&gt; The following objects are masked from 'package:BiocGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine, intersect, setdiff, union</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'tidyr'</span>
<span class="co">#&gt; The following object is masked from 'package:S4Vectors':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand</span></code></pre></div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<p>We start with the peptide-level output from Proteome Discoverer (PD). This experiment was designed to identify RNA-binding proteins (RBP) in the U-2 OS cell line using the OOPS method <span class="citation">(Queiroz et al. 2019)</span> with a comparison of RNase +/- used to separate RBPs from background non-specific proteins. 4 replicate experiments were performed, where the RNase +/- experiments were performed from the same OOPS interface. For each LFQ run, approximately the same quantity of peptides were injected, based on quantification of peptide concentration post trypsin digestion. This data is not published but the aim of the experiment is equivalent to Figure 2e in the original OOPS paper<span class="citation">(Queiroz et al. 2019)</span>.</p>
<p>The data we will use is available through this package. This is simply a <code>PeptideGroups.txt</code> file from PD which has been read into memory using <code>read.delim(&lt;PeptideGroups.txt&gt;")</code>, where <code>&lt;PeptideGroups.txt&gt;</code> is the path to the PeptideGroups file.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pep_data</span> <span class="op">&lt;-</span> <span class="va">lfq_all_oops_rnase_pep</span></code></pre></div>
<p>We can explore the structure of the input data using <code>str</code>. We see that we have a <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> with 7808 rows and 41 columns. The most important columns to us are:</p>
<ul>
<li>Sequence: The sequence of the peptide</li>
<li>Modifications: The detected peptide modifications, including variable, e.g induced modifications such as oxidation</li>
<li>Master.Protein.Accessions: The assigned master protein(s)</li>
<li>Abundance.F*.Sample: Columns with the peptide intensities</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">pep_data</span><span class="op">)</span>
<span class="co">#&gt; 'data.frame':    7808 obs. of  41 variables:</span>
<span class="co">#&gt;  $ Peptide.Groups.Peptide.Group.ID                      : int  3 103128 103062 103059 103056 103049 103048 103045 102936 102916 ...</span>
<span class="co">#&gt;  $ Checked                                              : chr  "False" "False" "False" "False" ...</span>
<span class="co">#&gt;  $ Confidence                                           : chr  "High" "High" "High" "High" ...</span>
<span class="co">#&gt;  $ Sequence                                             : chr  "AAAAAAAAAAAAAAAGAGAGAK" "LVKPGNQNTQVTEAWNK" "LVGSVNLFSDENVPR" "LVGSQEELASWGHEYVR" ...</span>
<span class="co">#&gt;  $ Modifications                                        : chr  "" "" "" "" ...</span>
<span class="co">#&gt;  $ Qvality.PEP                                          : num  4.27e-07 8.31e-04 1.36e-03 7.92e-04 1.87e-06 ...</span>
<span class="co">#&gt;  $ Qvality.q.value                                      : num  0.00026 0.00026 0.00026 0.00026 0.00026 ...</span>
<span class="co">#&gt;  $ Number.of.Protein.Groups                             : int  1 1 1 1 1 1 1 1 1 1 ...</span>
<span class="co">#&gt;  $ Number.of.Proteins                                   : int  1 1 1 1 1 1 1 1 1 1 ...</span>
<span class="co">#&gt;  $ Number.of.PSMs                                       : int  8 8 4 8 16 7 26 12 10 11 ...</span>
<span class="co">#&gt;  $ Master.Protein.Accessions                            : chr  "P55011" "Q9UQ80" "Q9BQG0" "Q13200" ...</span>
<span class="co">#&gt;  $ Protein.Accessions                                   : chr  "P55011" "Q9UQ80" "Q9BQG0" "Q13200" ...</span>
<span class="co">#&gt;  $ Number.of.Missed.Cleavages                           : int  0 0 0 0 0 1 0 0 0 1 ...</span>
<span class="co">#&gt;  $ Theo.MHplus.in.Da                                    : num  1597 1927 1646 1960 1515 ...</span>
<span class="co">#&gt;  $ Abundance.F17.Sample                                 : num  1927597 3768782 NA NA 6090638 ...</span>
<span class="co">#&gt;  $ Abundance.F18.Sample                                 : num  3540084 6682988 NA NA 12042683 ...</span>
<span class="co">#&gt;  $ Abundance.F19.Sample                                 : num  3466165 14879391 NA NA 19815360 ...</span>
<span class="co">#&gt;  $ Abundance.F20.Sample                                 : num  789214 1826370 NA NA 3435918 ...</span>
<span class="co">#&gt;  $ Abundance.F21.Sample                                 : num  1425166 NA NA NA 1736376 ...</span>
<span class="co">#&gt;  $ Abundance.F22.Sample                                 : num  2515468 1258701 NA NA 2838919 ...</span>
<span class="co">#&gt;  $ Abundance.F23.Sample                                 : num  2426854 1511282 NA NA 3298768 ...</span>
<span class="co">#&gt;  $ Abundance.F24.Sample                                 : num  1370120 819806 NA NA 1711976 ...</span>
<span class="co">#&gt;  $ Quan.Info                                            : chr  "" "" "NoQuanValues" "NoQuanValues" ...</span>
<span class="co">#&gt;  $ Found.in.Sample.in.S17.F17.Sample                    : chr  "High" "High" "High" "High" ...</span>
<span class="co">#&gt;  $ Found.in.Sample.in.S18.F18.Sample                    : chr  "High" "High" "High" "High" ...</span>
<span class="co">#&gt;  $ Found.in.Sample.in.S19.F19.Sample                    : chr  "High" "High" "Not Found" "High" ...</span>
<span class="co">#&gt;  $ Found.in.Sample.in.S20.F20.Sample                    : chr  "High" "High" "Not Found" "Not Found" ...</span>
<span class="co">#&gt;  $ Found.in.Sample.in.S21.F21.Sample                    : chr  "High" "Not Found" "Not Found" "Not Found" ...</span>
<span class="co">#&gt;  $ Found.in.Sample.in.S22.F22.Sample                    : chr  "High" "High" "Not Found" "Not Found" ...</span>
<span class="co">#&gt;  $ Found.in.Sample.in.S23.F23.Sample                    : chr  "High" "Peak Found" "Not Found" "High" ...</span>
<span class="co">#&gt;  $ Found.in.Sample.in.S24.F24.Sample                    : chr  "High" "Peak Found" "Not Found" "Not Found" ...</span>
<span class="co">#&gt;  $ Confidence.by.Search.Engine.MS.Amanda.20             : chr  "n/a" "High" "High" "High" ...</span>
<span class="co">#&gt;  $ Confidence.by.Search.Engine.Sequest.HT               : chr  "High" "High" "High" "High" ...</span>
<span class="co">#&gt;  $ Percolator.q.Value.by.Search.Engine.MS.Amanda.20     : num  NA 0.000358 0.000434 0.000347 0.000434 ...</span>
<span class="co">#&gt;  $ Percolator.q.Value.by.Search.Engine.Sequest.HT       : num  0.00041 0.000469 0.000457 0.00041 0.000469 ...</span>
<span class="co">#&gt;  $ Percolator.PEP.by.Search.Engine.MS.Amanda.20         : num  NA 3.76e-04 6.54e-04 3.47e-04 1.06e-06 ...</span>
<span class="co">#&gt;  $ Percolator.PEP.by.Search.Engine.Sequest.HT           : num  3.16e-07 3.91e-04 3.71e-04 1.21e-03 1.69e-06 ...</span>
<span class="co">#&gt;  $ Amanda.Score.by.Search.Engine.MS.Amanda.20           : num  NA 45.4 81.1 65.3 132.2 ...</span>
<span class="co">#&gt;  $ CharmeRT.Combined.Score.by.Search.Engine.MS.Amanda.20: num  NA 45.4 81.1 65.3 132.2 ...</span>
<span class="co">#&gt;  $ XCorr.by.Search.Engine.Sequest.HT                    : num  5.77 3 4.34 4.37 3.85 5.92 3.09 4.3 2.92 3.58 ...</span>
<span class="co">#&gt;  $ Top.Apex.RT.in.min                                   : num  65.7 57.8 NA NA 66.6 ...</span></code></pre></div>
<p>The samples analysed are as follows:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sample_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  File <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"F"</span>, <span class="fl">17</span><span class="op">:</span><span class="fl">24</span><span class="op">)</span>,
  Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RNase_neg"</span>, <span class="st">"RNase_pos"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>, <span class="st">"."</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span>,
  Condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RNase_neg"</span>, <span class="st">"RNase_pos"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>,
  Replicate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">2</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">File</th>
<th align="left">Sample</th>
<th align="left">Condition</th>
<th align="right">Replicate</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">F17</td>
<td align="left">RNase_neg.1</td>
<td align="left">RNase_neg</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">F18</td>
<td align="left">RNase_neg.2</td>
<td align="left">RNase_neg</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">F19</td>
<td align="left">RNase_neg.3</td>
<td align="left">RNase_neg</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">F20</td>
<td align="left">RNase_neg.4</td>
<td align="left">RNase_neg</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">F21</td>
<td align="left">RNase_pos.1</td>
<td align="left">RNase_pos</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">F22</td>
<td align="left">RNase_pos.2</td>
<td align="left">RNase_pos</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">F23</td>
<td align="left">RNase_pos.3</td>
<td align="left">RNase_pos</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">F24</td>
<td align="left">RNase_pos.4</td>
<td align="left">RNase_pos</td>
<td align="right">4</td>
</tr>
</tbody>
</table>
<div class="section level3">
<h3 id="parse-input-data-and-remove-contaminants">Parse input data and remove contaminants<a class="anchor" aria-label="anchor" href="#parse-input-data-and-remove-contaminants"></a>
</h3>
<p>Before we start to parse the input data, we need to identify the contaminant proteins, which were supplied to Proteome Discoverer. We want to remove these from the data. These were defined using the <a href="https://www.thegpm.org/crap/" class="external-link">cRAP database</a>. Below, we parse the cRAP FASTA to extract the IDs for the cRAP proteins, in both ‘cRAP’ format and UniProt accessions for these proteins to be completely sure we are remoing all cRAP.</p>
<p>Again, the file we will use <code>cRAP_20190401.fasta.gz</code> is available through the <code>camprotR</code> package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">crap_fasta_inf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>
  <span class="st">"extdata"</span>, <span class="st">"cRAP_20190401.fasta.gz"</span>, 
  package <span class="op">=</span> <span class="st">"camprotR"</span>
<span class="op">)</span>

<span class="co"># Load the cRAP FASTA used for the PD search</span>
<span class="va">crap_fasta</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">fasta.index</a></span><span class="op">(</span><span class="va">crap_fasta_inf</span>, seqtype <span class="op">=</span> <span class="st">"AA"</span><span class="op">)</span>

<span class="co"># Extract the UniProt accessions associated with each cRAP protein</span>
<span class="va">crap_accessions</span> <span class="op">&lt;-</span> <span class="va">crap_fasta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">desc</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_extract.html" class="external-link">str_extract_all</a></span><span class="op">(</span>pattern<span class="op">=</span><span class="st">"(?&lt;=\\|).*?(?=\\|)"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>We can then supply these cRAP protein IDs to <code><a href="https://rdrr.io/pkg/camprotR/man/parse_features.html" class="external-link">camprotR::parse_features()</a></code> which will remove features (i.e. peptides in this case) which may originate from contaminants, as well as features which don’t have a unique master protein.</p>
<p>See <code><a href="https://rdrr.io/pkg/camprotR/man/parse_features.html" class="external-link">?parse_features</a></code> for further details, including the removal of ‘associated cRAP’ for conservative contaminants removal.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pep_data_flt</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/parse_features.html" class="external-link">parse_features</a></span><span class="op">(</span>
  <span class="va">pep_data</span>,
  TMT <span class="op">=</span> <span class="cn">FALSE</span>, 
  level <span class="op">=</span> <span class="st">'peptide'</span>,
  crap_proteins <span class="op">=</span> <span class="va">crap_accessions</span>, 
<span class="op">)</span>
<span class="co">#&gt; Parsing features...</span>
<span class="co">#&gt; 7808 features found from 1463 master proteins =&gt; Input</span>
<span class="co">#&gt; 242 cRAP proteins supplied</span>
<span class="co">#&gt; 364 proteins identified as 'cRAP associated'</span>
<span class="co">#&gt; 7509 features found from 1414 master proteins =&gt; cRAP features removed</span>
<span class="co">#&gt; 7476 features found from 1396 master proteins =&gt; associated cRAP features removed</span>
<span class="co">#&gt; 7471 features found from 1395 master proteins =&gt; features without a master protein removed</span>
<span class="co">#&gt; 7250 features found from 1307 master proteins =&gt; features with non-unique master proteins removed</span></code></pre></div>
<p>From the above, we can see that we have started with 7808 ‘features’ (peptides) from 1463 master proteins across all samples. After removal of contaminants and peptides that can’t be assigned to a unique master protein, we have 7250 peptides remaining from 1307 master proteins.</p>
</div>
<div class="section level3">
<h3 id="convert-to-msnset">Convert to MSnSet<a class="anchor" aria-label="anchor" href="#convert-to-msnset"></a>
</h3>
<p>We now store the filtered peptide data in an MSnSet, the standard data object for proteomics in R. See the <code>vignette("msnset", package="camprotR")</code> for more details.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create expression matrix with peptide abundances (exprs) and human </span>
<span class="co"># readable column names</span>
<span class="va">exprs_data</span> <span class="op">&lt;-</span> <span class="va">pep_data_flt</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"Abundance"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">`colnames&lt;-`</span><span class="op">(</span><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Create data.frame with sample metadata (pData)</span>
<span class="va">pheno_data</span> <span class="op">&lt;-</span> <span class="va">sample_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">File</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"Sample"</span><span class="op">)</span>

<span class="co"># Create data.frame with peptide metadata (fData)</span>
<span class="va">feat_data</span> <span class="op">&lt;-</span> <span class="va">pep_data_flt</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"Abundance"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Create MSnSet</span>
<span class="va">pep</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnSet</a></span><span class="op">(</span>exprs <span class="op">=</span> <span class="va">exprs_data</span>,
                       fData <span class="op">=</span> <span class="va">feat_data</span>,
                       pData <span class="op">=</span> <span class="va">pheno_data</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="qc-peptides">QC peptides<a class="anchor" aria-label="anchor" href="#qc-peptides"></a>
</h3>
<p>First of all, we want to inspect the peptide intensity distributions. We expect these to be approximately equal and any very low intensity sample would be a concern that would need to be further explored. Here, we can see that there is some clear variability, but no sample with very low intensity.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pep</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span>base <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/plot_quant.html" class="external-link">plot_quant</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">'density'</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 19289 rows containing non-finite values (stat_density).</span></code></pre></div>
<div class="figure">
<img src="LFQ_files/figure-html/unnamed-chunk-8-1.png" alt="Peptide intensities" width="384"><p class="caption">
Peptide intensities
</p>
</div>
<p>Next, we consider the missing values. Note that <code><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">MSnbase::plotNA</a></code> assumes the object contains protein-level data and names the x-axis accordingly. Here, we update the plot aesthetics and rename the x-axis</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">plotNA</a></span><span class="op">(</span><span class="va">pep</span>, pNA <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Peptide index'</span><span class="op">)</span></code></pre></div>
<p>So, from the 7250 peptides, just 2085 have quantification values in all 8 samples. This is not a surprise for LFQ, since each sample is prepared and run separately.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="LFQ_files/figure-html/unnamed-chunk-10-1.png" alt="Data completeness, all peptides" width="480"><p class="caption">
Data completeness, all peptides
</p>
</div>
<p>We can explore the structure of the missing values further using an ‘upset’ plot.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">missing_data</span> <span class="op">&lt;-</span> <span class="va">pep</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu">naniar</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/naniar/man/gg_miss_upset.html" class="external-link">gg_miss_upset</a></span><span class="op">(</span><span class="va">missing_data</span>,
                      sets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pep</span><span class="op">)</span>, <span class="st">'_NA'</span><span class="op">)</span>,
                      keep.order <span class="op">=</span> <span class="cn">TRUE</span>,
                      nsets <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="LFQ_files/figure-html/unnamed-chunk-11-1.png" alt="Missing values upset plot" width="768"><p class="caption">
Missing values upset plot
</p>
</div>
<p>So in this case, we can see that the most common missing value patterns are:</p>
<ol style="list-style-type: decimal">
<li>Missing in just RNase negative replicate 4</li>
<li>Missing in all samples.</li>
<li>Missing in all the other samples, except RNase negative replicate 4</li>
</ol>
<p>RNase negative replicate 4 had slightly lower overall petide intensities and appears to be somewhat of an outlier. In this case, we will retain the sample but in other cases, this may warrant further exploration and potentially removal of a sample.</p>
<p>Finally, we recode the abundances to binary (present/absent) and plot the missing data structure as a heatmap (missing = black). Beyond the aforementioned observations regarding RNase negative replicate 4, there’s no clear structure in the missing values. This is what we would expect for LFQ, where data is largely missing at random (MAR). There are methods to impute values MAR, but, providing what we are after is protein-level abundance, it is usually more appropriate to perform the protein inference using a method that can accommodate missing values.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">missing_data</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">missing_data</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="va">missing_data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">missing_data</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span>

<span class="fu">gplots</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gplots/man/heatmap.2.html" class="external-link">heatmap.2</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">missing_data</span><span class="op">)</span>,
  col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"lightgray"</span><span class="op">)</span>,
  scale <span class="op">=</span> <span class="st">"none"</span>, <span class="co"># Don't re-scale the input</span>
  <span class="co"># Default is euclidean, which is less appropriate distance for binary vectors</span>
  distfun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">'binary'</span><span class="op">)</span>, 
  trace <span class="op">=</span> <span class="st">"none"</span>, <span class="co"># Don't include a trace. Improves appearance</span>
  key <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Don't include a key as values are binary</span>
  Colv <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Don't re-order the columns</span>
  labRow <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Don't label the rows</span>
  cexCol <span class="op">=</span> <span class="fl">0.7</span> <span class="co"># Reduce the column name size</span>
<span class="op">)</span>
<span class="co">#&gt; Warning in gplots::heatmap.2(as.matrix(missing_data), col = c("black",</span>
<span class="co">#&gt; "lightgray"), : Discrepancy: Colv is FALSE, while dendrogram is `both'. Omitting</span>
<span class="co">#&gt; column dendogram.</span></code></pre></div>
<div class="figure">
<img src="LFQ_files/figure-html/unnamed-chunk-12-1.png" alt="Presence/Absence heatmap" width="576"><p class="caption">
Presence/Absence heatmap
</p>
</div>
</div>
<div class="section level3">
<h3 id="normalise-peptide-intensities">Normalise peptide intensities<a class="anchor" aria-label="anchor" href="#normalise-peptide-intensities"></a>
</h3>
<p>We have injected the same quantity of peptides for each sample, so it’s reasonable to normalise the samples against one another. Here we will apply ‘center-median’ normalisation, which in <code><a href="https://lgatto.github.io/MSnbase/reference/normalise-methods.html" class="external-link">MSnbase::normalise</a></code> is called ‘diff.median’. Since the peptide intensities are log-Gaussian distributed, we log<sub>2</sub>-transform them before performing the normalisation.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pep_norm</span> <span class="op">&lt;-</span> <span class="va">pep</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span>base <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/normalise-methods.html" class="external-link">normalise</a></span><span class="op">(</span><span class="st">'diff.median'</span><span class="op">)</span>  

<span class="va">pep_norm</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/plot_quant.html" class="external-link">plot_quant</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">'density'</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 19289 rows containing non-finite values (stat_density).</span></code></pre></div>
<div class="figure">
<img src="LFQ_files/figure-html/unnamed-chunk-13-1.png" alt="Protein intensities post-normalisation" width="480"><p class="caption">
Protein intensities post-normalisation
</p>
</div>
</div>
<div class="section level3">
<h3 id="summarising-to-protein-level-abundance">Summarising to protein-level abundance<a class="anchor" aria-label="anchor" href="#summarising-to-protein-level-abundance"></a>
</h3>
<p>Before we can summarise to protein-level abundances, we need to exclude peptides with too many missing values. Here, peptides with more than 5/8 missing values are discarded, using <code><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnbase::filterNA()</a></code>. We also need to remove proteins without at least two peptides. We will use <code><a href="https://rdrr.io/pkg/camprotR/man/restrict_features_per_protein.html" class="external-link">camprotR::restrict_features_per_protein()</a></code> which will replace quantification values with <code>NA</code> if the sample does not have two quantified peptides for a given protein. Note that this means we have to repeat the filtering process since we are adding missing values.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pep_restricted</span> <span class="op">&lt;-</span> <span class="va">pep_norm</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">filterNA</a></span><span class="op">(</span>pNA <span class="op">=</span> <span class="fl">4</span><span class="op">/</span><span class="fl">8</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># Maximum 4/8 missing values</span>
  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/restrict_features_per_protein.html" class="external-link">restrict_features_per_protein</a></span><span class="op">(</span>min_features <span class="op">=</span> <span class="fl">2</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># At least two peptides per protein</span>
  <span class="co"># Repeat the filtering since restrict_features_per_protein will replace some values with NA</span>
  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">filterNA</a></span><span class="op">(</span>pNA <span class="op">=</span> <span class="fl">4</span><span class="op">/</span><span class="fl">8</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/restrict_features_per_protein.html" class="external-link">restrict_features_per_protein</a></span><span class="op">(</span>min_features <span class="op">=</span> <span class="fl">2</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </code></pre></div>
<p>We can then re-inspect the missing values. Note that we have reduced the overall number of peptides to 4971.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">plotNA</a></span><span class="op">(</span><span class="va">pep_restricted</span>, pNA <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Peptide index'</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="LFQ_files/figure-html/unnamed-chunk-16-1.png" alt="Data completeness, retained peptides" width="576"><p class="caption">
Data completeness, retained peptides
</p>
</div>
<p>We can now summarise to protein-level abundance. Below, we use ‘robust’ summarisation <span class="citation">(Sticker et al. 2020)</span> with <code><a href="https://lgatto.github.io/MSnbase/reference/combineFeatures.html" class="external-link">MSnbase::combineFeatures()</a></code>. This returns a warning about missing values that we can ignore here since the robust method is inherently designed to handle missing values. See <code><a href="https://rdrr.io/pkg/MsCoreUtils/man/robustSummary.html" class="external-link">MsCoreUtils::robustSummary()</a></code> and <a href="https://pubmed.ncbi.nlm.nih.gov/32321741/" class="external-link">this publication</a> for further details about the robust method.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prot_robust</span> <span class="op">&lt;-</span> <span class="va">pep_restricted</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/combineFeatures.html" class="external-link">combineFeatures</a></span><span class="op">(</span>
    <span class="co"># group the peptides by their master protein id</span>
    groupBy <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_restricted</span><span class="op">)</span><span class="op">$</span><span class="va">Master.Protein.Accessions</span>, 
    method <span class="op">=</span> <span class="st">'robust'</span>, 
    maxit <span class="op">=</span> <span class="fl">1000</span>  <span class="co"># Ensures convergence for MASS::rlm</span>
  <span class="op">)</span> 
<span class="co">#&gt; Your data contains missing values. Please read the relevant section in</span>
<span class="co">#&gt; the combineFeatures manual page for details on the effects of missing</span>
<span class="co">#&gt; values on data aggregation.</span></code></pre></div>
<p>We can then re-inspect the missing values at the protein level. So, we have quantification for 681 proteins, of which 409 are fully quantified across all 8 samples. The most common missing values pattern remains missing in just RNase negative replicate 4.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">plotNA</a></span><span class="op">(</span><span class="va">prot_robust</span>, pNA <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<p><img src="LFQ_files/figure-html/unnamed-chunk-19-1.png" width="576"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">naniar</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/naniar/man/gg_miss_upset.html" class="external-link">gg_miss_upset</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span><span class="op">)</span>,
                      sets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span>, <span class="st">'_NA'</span><span class="op">)</span>,
                      keep.order <span class="op">=</span> <span class="cn">TRUE</span>,
                      nsets <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<p><img src="LFQ_files/figure-html/unnamed-chunk-20-1.png" width="768"></p>
<p>We have now obtained protein-level abundances from our LFQ data. This is our recommended procedure for summarising from peptide to protein level abundances.</p>
</div>
<div class="section level3">
<h3 id="alternatives-for-summarising-to-protein-level-abundance---maxlfq">Alternatives for summarising to protein-level abundance - MaxLFQ<a class="anchor" aria-label="anchor" href="#alternatives-for-summarising-to-protein-level-abundance---maxlfq"></a>
</h3>
<p>Although we recommend using the ‘robust’ summarisation, there are other algorithms that are commonly applied. For example MaxLFQ <span class="citation">(Cox et al. 2014)</span>, which is implemented within MaxQuant, but also available via the <code><a href="https://rdrr.io/pkg/iq/man/maxLFQ.html" class="external-link">iq::maxLFQ()</a></code> function in the iq package in R (now available on <a href="https://cran.r-project.org/web/packages/iq/index.html" class="external-link">CRAN</a>).</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># You may wish to retain more feature columns that this! </span>
<span class="va">feature_coloumns_to_retain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Master.Protein.Accessions"</span><span class="op">)</span>

<span class="co"># To ensure we are only comparing the protein inference method and not normalisation etc,</span>
<span class="co"># we will use the same set of peptides for both inference methods</span>
<span class="va">pep_data_for_summarisation</span> <span class="op">&lt;-</span> <span class="va">pep_restricted</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_restricted</span><span class="op">)</span><span class="op">[</span>, <span class="va">feature_coloumns_to_retain</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, 
        by <span class="op">=</span> <span class="st">'row.names'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">Row.names</span><span class="op">)</span></code></pre></div>
<p>Below, we use <code><a href="https://rdrr.io/pkg/iq/man/maxLFQ.html" class="external-link">iq::maxLFQ()</a></code> manually, though it should be possible to use this function within <code><a href="https://lgatto.github.io/MSnbase/reference/combineFeatures.html" class="external-link">MSnbase::combineFeatures()</a></code> too in theory since it accommodates user-defined functions.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Define a function to perform MaxLFQ on a single protein and return a data.frame</span>
<span class="co"># as required. </span>
<span class="va">get_maxlfq_estimate</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">prot</span> <span class="op">&lt;-</span> <span class="fu">iq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/iq/man/maxLFQ.html" class="external-link">maxLFQ</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">estimate</span>
  
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">prot</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># Group by the features we want to retain and use MaxLFQ on each protein</span>
<span class="va">maxlfq_estimates</span> <span class="op">&lt;-</span> <span class="va">pep_data_for_summarisation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">all_of</a></span><span class="op">(</span><span class="va">feature_coloumns_to_retain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_map.html" class="external-link">group_modify</a></span><span class="op">(</span><span class="op">~</span> <span class="fu">get_maxlfq_estimate</span><span class="op">(</span><span class="va">.</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Create the protein-level MSnSet</span>
<span class="va">maxlfq.e</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">maxlfq_estimates</span>, <span class="op">-</span><span class="va">Master.Protein.Accessions</span><span class="op">)</span><span class="op">)</span>
<span class="va">maxlfq.f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">maxlfq_estimates</span>, <span class="va">Master.Protein.Accessions</span><span class="op">)</span><span class="op">)</span>
<span class="va">maxlfq.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">pep_restricted</span><span class="op">)</span>

<span class="va">prot_maxlfq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnSet</a></span><span class="op">(</span>exprs <span class="op">=</span> <span class="va">maxlfq.e</span>,
                      fData <span class="op">=</span> <span class="va">maxlfq.f</span>,
                      pData <span class="op">=</span> <span class="va">maxlfq.p</span><span class="op">)</span>

<span class="co"># Update the rownames to be the protein IDs</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_maxlfq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">maxlfq_estimates</span><span class="op">$</span><span class="va">Master.Protein.Accessions</span></code></pre></div>
<p>We can now compare the protein-level abundance estimates.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Define a function to extract the protein abundances in long form and</span>
<span class="co"># add a column annotating the method</span>
<span class="va">get_long_form_prot_exp</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">obj</span>, <span class="va">method_name</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://generics.r-lib.org/reference/tidy.html" class="external-link">tidy</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>method <span class="op">=</span> <span class="va">method_name</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># Single object with protein inference from both methods </span>
<span class="va">compare_protein_abundances</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu">get_long_form_prot_exp</span><span class="op">(</span><span class="va">prot_maxlfq</span>, <span class="st">'MaxLFQ'</span><span class="op">)</span>,
  <span class="fu">get_long_form_prot_exp</span><span class="op">(</span><span class="va">prot_robust</span>, <span class="st">'Robust'</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># Plot direct comparison</span>
<span class="va">compare_protein_abundances</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">method</span>, values_from <span class="op">=</span> <span class="va">value</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># pivot to wider form for plotting</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Robust</span>, y <span class="op">=</span> <span class="va">MaxLFQ</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.25</span>, size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, colour <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_cat_palette.html" class="external-link">get_cat_palette</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, linetype <span class="op">=</span> <span class="fl">2</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="LFQ_files/figure-html/unnamed-chunk-23-1.png" alt="Comparison of protein inference methods" width="480"><p class="caption">
Comparison of protein inference methods
</p>
</div>
<p>There is a very good overall correlation. Let’s inspect a few proteins with the largest differences between the two approaches to see what’s going on for the edge cases.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Identify proteins with largest difference between the protein summarisation methods</span>
<span class="va">proteins_of_interest</span> <span class="op">&lt;-</span> <span class="va">compare_protein_abundances</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">method</span>, values_from <span class="op">=</span> <span class="va">value</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>diff <span class="op">=</span> <span class="va">MaxLFQ</span><span class="op">-</span><span class="va">Robust</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">diff</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
<p>Below we define a function to plot the peptide and protein abundances for the two methods for a single protein. We can ignore the details since it’s the plots themselves we are interested in.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_pep_and_protein</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">protein_of_interest</span><span class="op">)</span> <span class="op">{</span>
  
  <span class="va">to_plot_compare</span> <span class="op">&lt;-</span> <span class="va">compare_protein_abundances</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein</span> <span class="op">==</span> <span class="va">protein_of_interest</span><span class="op">)</span>
  
  <span class="va">pep_restricted</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_restricted</span><span class="op">)</span><span class="op">$</span><span class="va">Master.Protein.Accession</span> <span class="op">==</span> <span class="va">protein_of_interest</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">'id'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="va">id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">name</span>, y <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group <span class="op">=</span> <span class="va">id</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">'grey'</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">'grey'</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">to_plot_compare</span>,
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample.id</span>, y <span class="op">=</span> <span class="va">value</span>, colour <span class="op">=</span> <span class="va">method</span>, group <span class="op">=</span> <span class="va">method</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">to_plot_compare</span>,
               <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample.id</span>, y <span class="op">=</span> <span class="va">value</span>, colour <span class="op">=</span> <span class="va">method</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_cat_palette.html" class="external-link">get_cat_palette</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">'LFQ summarisation method'</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">15</span>, border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>
      title <span class="op">=</span> <span class="va">protein_of_interest</span>,
      x <span class="op">=</span> <span class="st">''</span>,
      y <span class="op">=</span> <span class="st">'Protein abundance (log2)'</span>
    <span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>Below we apply the above function to each of our proteins of interest.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proteins_of_interest</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">plot_pep_and_protein</span><span class="op">)</span>
<span class="co">#&gt; [[1]]</span></code></pre></div>
<p><img src="LFQ_files/figure-html/unnamed-chunk-26-1.png" width="672"></p>
<pre><code><span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span></code></pre>
<p><img src="LFQ_files/figure-html/unnamed-chunk-26-2.png" width="672"></p>
<pre><code><span class="co">#&gt; </span>
<span class="co">#&gt; [[3]]</span></code></pre>
<p><img src="LFQ_files/figure-html/unnamed-chunk-26-3.png" width="672"></p>
<pre><code><span class="co">#&gt; </span>
<span class="co">#&gt; [[4]]</span></code></pre>
<p><img src="LFQ_files/figure-html/unnamed-chunk-26-4.png" width="672"></p>
<pre><code><span class="co">#&gt; </span>
<span class="co">#&gt; [[5]]</span></code></pre>
<p><img src="LFQ_files/figure-html/unnamed-chunk-26-5.png" width="672"></p>
<p>Looking at these examples, we can see that MaxLFQ is often estimating slightly higher abundances but with a very similar profile across the samples is very similar, so the summarisation approach is unlikely to affect the downstream analysis. It’s not clear which of the two approaches is more correct in the examples above, but the publication proposing the robust protein inference (see <a href="https://pubmed.ncbi.nlm.nih.gov/32321741/" class="external-link">here</a>) does indicate it gives more accurate fold-change estimates overall.</p>
<p>We have now processed our peptide-level LFQ abundances and obtained protein-level abundances, from which we can perform our downstream analyses.</p>
</div>
</div>
<div class="section level2">
<h2 id="tom---move-this-section-to-an-optional-extension-notebook">TOM - MOVE THIS SECTION TO AN OPTIONAL ‘EXTENSION’ notebook<a class="anchor" aria-label="anchor" href="#tom---move-this-section-to-an-optional-extension-notebook"></a>
</h2>
<div class="section level3">
<h3 id="alternative-normalisation---using-reference-proteins">Alternative normalisation - Using reference proteins<a class="anchor" aria-label="anchor" href="#alternative-normalisation---using-reference-proteins"></a>
</h3>
<p>Finally, we consider an alternative normalisation approach where we have a strong prior expectation about the abundance of a subset of proteins. Our data is from an RNase +/- OOPS experiment. Since OOPS is known to enrich glycoproteins at the interface, we can use these proteins as an internal set of ‘housekeeping’ proteins which we expect to have no difference in abundance between RNase +/-.</p>
<p>First, we need to calculate the RNase +/- ratios, since this is the value we want to normalise. In other applications, you may wish to normalise the abundance estimates for each sample, rather than the ratio between samples.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ratios</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span><span class="op">$</span><span class="va">Condition</span> <span class="op">==</span> <span class="st">'RNase_neg'</span><span class="op">]</span><span class="op">)</span> <span class="op">-</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span><span class="op">$</span><span class="va">Condition</span> <span class="op">==</span> <span class="st">'RNase_pos'</span><span class="op">]</span><span class="op">)</span>

<span class="va">prot_ratios</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnSet</a></span><span class="op">(</span>exprs <span class="op">=</span> <span class="va">ratios</span>,
                      fData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span>,
                      pData <span class="op">=</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Condition</span> <span class="op">==</span> <span class="st">'RNase_neg'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Next, we need annotations regarding which proteins are glycoproteins and GO-annotated RBPs. [Once a notebook has been added to cover GO annotations, reference to that here to avoid repeating explanations about why we want to expand the set of GO anntotations]</p>
<p>Get all GO terms for our proteins by querying UniProt programmatically.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up_url</span> <span class="op">&lt;-</span> <span class="st">"https://www.uniprot.org/uploadlists/"</span>

<span class="va">request</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  query <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">" "</span><span class="op">)</span>,
  from <span class="op">=</span> <span class="st">"ACC+ID"</span>,
  to <span class="op">=</span> <span class="st">"ACC"</span>,
  format <span class="op">=</span> <span class="st">"tab"</span>,
  columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"id"</span>, <span class="st">"features"</span>, <span class="st">"go-id"</span>, sep <span class="op">=</span> <span class="st">","</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">response</span> <span class="op">&lt;-</span> <span class="fu">httr</span><span class="fu">::</span><span class="fu"><a href="https://httr.r-lib.org/reference/GET.html" class="external-link">GET</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">up_url</span>, query <span class="op">=</span> <span class="va">request</span><span class="op">)</span>

<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">httr</span><span class="fu">::</span><span class="fu"><a href="https://httr.r-lib.org/reference/content.html" class="external-link">content</a></span><span class="op">(</span><span class="va">response</span>, type <span class="op">=</span> <span class="st">"text"</span>, encoding <span class="op">=</span> <span class="st">"UTF8"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span>text <span class="op">=</span> <span class="va">.</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Entry</span>, <span class="va">Features</span>, <span class="va">Gene.ontology.IDs</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">go_long</span> <span class="op">&lt;-</span> <span class="va">res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate_rows.html" class="external-link">separate_rows</a></span><span class="op">(</span><span class="va">Gene.ontology.IDs</span>, sep <span class="op">=</span> <span class="st">"; "</span><span class="op">)</span>

<span class="co"># This step may take a minute or two</span>
<span class="va">go_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_ancestor_go.html" class="external-link">get_ancestor_go</a></span><span class="op">(</span>
  <span class="va">go_long</span>, 
  feature_col <span class="op">=</span> <span class="st">"Entry"</span>,
  go_col <span class="op">=</span> <span class="st">"Gene.ontology.IDs"</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; [1] "Getting all ancestor GO terms for 4396 observed terms. This may take a while!"</span>
<span class="co">#&gt; Expanding GO terms to include all ancestors for all entries</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get RBPs</span>
<span class="va">go_rbp</span> <span class="op">&lt;-</span> <span class="st">"GO:0003723"</span>

<span class="va">rbps</span> <span class="op">&lt;-</span> <span class="va">go_all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Gene.ontology.IDs</span> <span class="op">==</span> <span class="va">go_rbp</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">Entry</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get glycoproteins</span>
<span class="va">glycoproteins</span> <span class="op">&lt;-</span> <span class="va">res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Glycosylation"</span>, <span class="va">Features</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glyco.features <span class="op">=</span> <span class="va">Features</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate_rows.html" class="external-link">separate_rows</a></span><span class="op">(</span><span class="va">Glyco.features</span>, sep <span class="op">=</span> <span class="st">"; "</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Glycosylation"</span>, <span class="va">Glyco.features</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glyco.features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\(|\\)"</span>, <span class="st">""</span>, <span class="va">Glyco.features</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">Glyco.features</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="st">"Glycosylation.count"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">" "</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Entry</span>, <span class="va">Glycosylation.count</span><span class="op">)</span></code></pre></div>
<p>Add feature columns describing the glycoprotein and GO-RBP status of the proteins.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glycoprotein <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">glycoproteins</span><span class="op">$</span><span class="va">Entry</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>GO.RBP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">rbps</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glyco.RBP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interaction.html" class="external-link">interaction</a></span><span class="op">(</span><span class="va">Glycoprotein</span>, <span class="va">GO.RBP</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glyco.RBP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/recode.html" class="external-link">recode</a></span><span class="op">(</span>
    <span class="va">Glyco.RBP</span>,
    <span class="st">'TRUE.TRUE'</span><span class="op">=</span><span class="st">'GO:RBGP'</span>,
    <span class="st">'FALSE.TRUE'</span><span class="op">=</span><span class="st">'GO:RBP'</span>,
    <span class="st">'TRUE.FALSE'</span><span class="op">=</span><span class="st">'Glycoprotein'</span>,
    <span class="st">'FALSE.FALSE'</span><span class="op">=</span><span class="st">'Other'</span><span class="op">)</span>,
    levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'GO:RBP'</span>, <span class="st">'GO:RBGP'</span>, <span class="st">'Other'</span>, <span class="st">'Glycoprotein'</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<p>Below, we define a function to plot the ratios for each functional sub-type of proteins.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_ratios</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>,
    <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">[</span>,<span class="st">'Glyco.RBP'</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,
    by <span class="op">=</span> <span class="st">'row.names'</span>
  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Row.names</span>, <span class="va">Glyco.RBP</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">'sample'</span>, values_to <span class="op">=</span> <span class="st">'ratio'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>, by.x <span class="op">=</span> <span class="st">'sample'</span>, by.y <span class="op">=</span> <span class="st">'row.names'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">ratio</span><span class="op">)</span><span class="op">)</span>
  
  <span class="va">p</span> <span class="op">&lt;-</span> <span class="va">to_plot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Replicate</span>, y <span class="op">=</span> <span class="va">ratio</span>, 
               group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interaction.html" class="external-link">interaction</a></span><span class="op">(</span><span class="va">Glyco.RBP</span>, <span class="va">Replicate</span><span class="op">)</span>, 
               colour <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">Glyco.RBP</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_cat_palette.html" class="external-link">get_cat_palette</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, <span class="st">'black'</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">''</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, linetype <span class="op">=</span> <span class="fl">2</span>, colour <span class="op">=</span> <span class="st">'grey'</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>
      x <span class="op">=</span> <span class="st">"Replicate"</span>,
      y <span class="op">=</span> <span class="st">"RNase -/+ ratio"</span>
    <span class="op">)</span>
  
  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
  
  <span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="va">to_plot</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>How do the protein ratios look pre-normalisation…</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plot_ratios</span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span></code></pre></div>
<p><img src="LFQ_files/figure-html/unnamed-chunk-34-1.png" width="768"></p>
<p>OK, so the glycoproteins are not centered at zero and there are GO-annotated RBPs with negative log RNase -/+ ratios (as much as ~25% in replicate 2)</p>
<p>Below, we perform the center median normalisation to the set of reference proteins (here, the glycoproteins).</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">glycoprotein_medians</span> <span class="op">&lt;-</span> <span class="va">prot_ratios</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span><span class="op">$</span><span class="va">Glyco.RBP</span> <span class="op">==</span> <span class="st">'Glycoprotein'</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_medians.html" class="external-link">get_medians</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">prot_ratios_norm</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/center_normalise_to_ref.html" class="external-link">center_normalise_to_ref</a></span><span class="op">(</span>
  <span class="va">prot_ratios</span>,
  <span class="va">glycoprotein_medians</span>,
  center_to_zero <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># We want to center the glycoproteins around zero</span>
  on_log_scale <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># The quantifications are on a log scale (log2 ratios)</span>
<span class="op">)</span></code></pre></div>
<p>And plot the protein ratios post-normalisation.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plot_ratios</span><span class="op">(</span><span class="va">prot_ratios_norm</span><span class="op">)</span></code></pre></div>
<p><img src="LFQ_files/figure-html/unnamed-chunk-36-1.png" width="768"></p>
<p>Now, the median log2 RNase -/+ ratio for glycoproteins is zero for all replicates and we have far fewer GO-annotated RBPs with negative log RNase -/+ ratios.</p>
</div>
<div class="section level3">
<h3 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<pre><code><span class="co">#&gt; R version 4.1.3 (2022-03-10)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.4 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span>
<span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span>
<span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] tidyr_1.2.0                    dplyr_1.0.8                   </span>
<span class="co">#&gt;  [3] Proteomics.data.analysis_0.1.0 camprotR_0.0.0.9000           </span>
<span class="co">#&gt;  [5] biobroom_1.26.0                broom_0.8.0                   </span>
<span class="co">#&gt;  [7] MSnbase_2.20.4                 ProtGenerics_1.26.0           </span>
<span class="co">#&gt;  [9] S4Vectors_0.32.4               mzR_2.28.0                    </span>
<span class="co">#&gt; [11] Rcpp_1.0.8.3                   Biobase_2.54.0                </span>
<span class="co">#&gt; [13] BiocGenerics_0.40.0            ggplot2_3.3.5                 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] colorspace_2.0-3       ellipsis_0.3.2         visdat_0.5.3          </span>
<span class="co">#&gt;   [4] rprojroot_2.0.3        XVector_0.34.0         fs_1.5.2              </span>
<span class="co">#&gt;   [7] clue_0.3-60            farver_2.1.0           affyio_1.64.0         </span>
<span class="co">#&gt;  [10] bit64_4.0.5            AnnotationDbi_1.56.2   fansi_1.0.3           </span>
<span class="co">#&gt;  [13] codetools_0.2-18       ncdf4_1.19             doParallel_1.0.17     </span>
<span class="co">#&gt;  [16] cachem_1.0.6           impute_1.68.0          robustbase_0.95-0     </span>
<span class="co">#&gt;  [19] knitr_1.38             jsonlite_1.8.0         naniar_0.6.1          </span>
<span class="co">#&gt;  [22] GO.db_3.14.0           cluster_2.1.3          vsn_3.62.0            </span>
<span class="co">#&gt;  [25] png_0.1-7              BiocManager_1.30.16    compiler_4.1.3        </span>
<span class="co">#&gt;  [28] httr_1.4.2             backports_1.4.1        assertthat_0.2.1      </span>
<span class="co">#&gt;  [31] fastmap_1.1.0          limma_3.50.3           cli_3.2.0             </span>
<span class="co">#&gt;  [34] htmltools_0.5.2        tools_4.1.3            gtable_0.3.0          </span>
<span class="co">#&gt;  [37] glue_1.6.2             GenomeInfoDbData_1.2.7 affy_1.72.0           </span>
<span class="co">#&gt;  [40] MALDIquant_1.21        jquerylib_0.1.4        pkgdown_2.0.2         </span>
<span class="co">#&gt;  [43] vctrs_0.4.1            Biostrings_2.62.0      preprocessCore_1.56.0 </span>
<span class="co">#&gt;  [46] iterators_1.0.14       xfun_0.30              stringr_1.4.0         </span>
<span class="co">#&gt;  [49] lifecycle_1.0.1        iq_1.9.3               gtools_3.9.2          </span>
<span class="co">#&gt;  [52] XML_3.99-0.9           DEoptimR_1.0-11        zlibbioc_1.40.0       </span>
<span class="co">#&gt;  [55] MASS_7.3-56            scales_1.2.0           ragg_1.2.2            </span>
<span class="co">#&gt;  [58] pcaMethods_1.86.0      parallel_4.1.3         yaml_2.3.5            </span>
<span class="co">#&gt;  [61] curl_4.3.2             memoise_2.0.1          gridExtra_2.3         </span>
<span class="co">#&gt;  [64] UpSetR_1.4.0           sass_0.4.1             stringi_1.7.6         </span>
<span class="co">#&gt;  [67] RSQLite_2.2.12         highr_0.9              desc_1.4.1            </span>
<span class="co">#&gt;  [70] foreach_1.5.2          caTools_1.18.2         BiocParallel_1.28.3   </span>
<span class="co">#&gt;  [73] GenomeInfoDb_1.30.1    rlang_1.0.2            pkgconfig_2.0.3       </span>
<span class="co">#&gt;  [76] systemfonts_1.0.4      bitops_1.0-7           mzID_1.32.0           </span>
<span class="co">#&gt;  [79] evaluate_0.15          lattice_0.20-45        purrr_0.3.4           </span>
<span class="co">#&gt;  [82] labeling_0.4.2         bit_4.0.4              tidyselect_1.1.2      </span>
<span class="co">#&gt;  [85] plyr_1.8.7             magrittr_2.0.3         R6_2.5.1              </span>
<span class="co">#&gt;  [88] IRanges_2.28.0         gplots_3.1.1           generics_0.1.2        </span>
<span class="co">#&gt;  [91] DBI_1.1.2              pillar_1.7.0           withr_2.5.0           </span>
<span class="co">#&gt;  [94] MsCoreUtils_1.6.2      KEGGREST_1.34.0        RCurl_1.98-1.6        </span>
<span class="co">#&gt;  [97] tibble_3.1.6           crayon_1.5.1           KernSmooth_2.23-20    </span>
<span class="co">#&gt; [100] utf8_1.2.2             rmarkdown_2.13         grid_4.1.3            </span>
<span class="co">#&gt; [103] blob_1.2.3             digest_0.6.29          textshaping_0.3.6     </span>
<span class="co">#&gt; [106] munsell_0.5.0          bslib_0.3.1</span></code></pre>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references">
<div id="ref-http://zotero.org/users/5634351/items/87YIDK69">
<p>Cox, Jürgen, Marco Y. Hein, Christian A. Luber, Igor Paron, Nagarjuna Nagaraj, and Matthias Mann. 2014. “Accurate Proteome-Wide Label-Free Quantification by Delayed Normalization and Maximal Peptide Ratio Extraction, Termed MaxLFQ*.” <em>Molecular &amp; Cellular Proteomics</em> 13 (9): 2513–26. <a href="https://doi.org/10.1074/mcp.M113.031591" class="external-link">https://doi.org/10.1074/mcp.M113.031591</a>.</p>
</div>
<div id="ref-http://zotero.org/users/5634351/items/EJHWF46N">
<p>Queiroz, Rayner M. L., Tom Smith, Eneko Villanueva, Maria Marti-Solano, Mie Monti, Mariavittoria Pizzinga, Dan-Mircea Mirea, et al. 2019. “Comprehensive Identification of RNA–Protein Interactions in Any Organism Using Orthogonal Organic Phase Separation (OOPS).” <em>Nature Biotechnology</em> 37 (2): 169. <a href="https://doi.org/10.1038/s41587-018-0001-2" class="external-link">https://doi.org/10.1038/s41587-018-0001-2</a>.</p>
</div>
<div id="ref-http://zotero.org/users/5634351/items/FZN3QTTZ">
<p>Sticker, Adriaan, Ludger Goeminne, Lennart Martens, and Lieven Clement. 2020. “Robust Summarization and Inference in Proteome-Wide Label-Free Quantification.” <em>Molecular &amp; Cellular Proteomics: MCP</em> 19 (7): 1209–19. <a href="https://doi.org/10.1074/mcp.RA119.001624" class="external-link">https://doi.org/10.1074/mcp.RA119.001624</a>.</p>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tom Smith.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
