<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Proteomics.data.analysis">
<title>Processing and QC of SILAC data • Proteomics.data.analysis</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Processing and QC of SILAC data">
<meta property="og:description" content="Proteomics.data.analysis">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Proteomics.data.analysis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/LFQ.html">Label-free Quantification Proteomics</a>
    <a class="dropdown-item" href="../articles/SILAC.html">Processing and QC of SILAC data</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="SILAC_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Processing and QC of SILAC data</h1>
                        <h4 data-toc-skip class="author">Tom Smith</h4>
            
            <h4 data-toc-skip class="date">2022-05-03</h4>
      
      
      <div class="d-none name"><code>SILAC.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="load-dependencies">Load dependencies<a class="anchor" aria-label="anchor" href="#load-dependencies"></a>
</h3>
<p>Load the required libraries.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mrctoxbioinformatics.github.io/Proteomics_data_analysis/">Proteomics.data.analysis</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">camprotR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://lgatto.github.io/MSnbase" class="external-link">MSnbase</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DEqMS</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">limma</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preamble">Preamble<a class="anchor" aria-label="anchor" href="#preamble"></a>
</h2>
<p>Stable Isotope Labelling by/with Amino acids in Cell culture (SILAC) is a form of quantitative proteomics where different conditions are quantified in the same run by differential metabolic labelling of proteins using amino acids containing stable isotopes <span class="citation">(Ong et al. 2002)</span>. A typical SILAC experiment involved growing cells in two different types of SILAC media, one containing ‘light’ arginine and lysine, and the other containing ‘heavy’ arginine and lysine. Over time the cells incorporate these amino acids into their proteins, changing the mass of the peptides detected by LC-MS. SILAC was initially designed to provide pairwise comparisons between cell cultures, but has now been extended to more than two labels and even whole organisms <span class="citation">(Krüger et al. 2008)</span>.</p>
<p>This elegant experimental design enables quantification of peptide/protein abundance ratios between conditions with very little technical variation, since the samples from separate conditions are pooled as early as possible in the sample handling process. For example, cell cultures can be treated with drug/control and then collected and pooled together for all downstream protocol steps. The use of different isotope labels has been extended to study the protein turnover (e.g switch from one label to another) in pulsed SILAC, and relative turnover between conditions (e.g two condition on the same label and then switch each condition to a different label; requires triple SILAC).</p>
<p>The analysis of SILAC data is relatively straightforward, since technical noise is low and normalisation is not normally required. Typically, one is interested in the intensity (peak area or peak height depending on your PD settings) ratio of the light (L) and heavy (H) MS peaks for a given peptide. This ratio forms the quantification value which we wish to perform statistical tests and exploratory analysis on.</p>
</div>
<div class="section level2">
<h2 id="asssessing-the-silac-isotope-incorporation-rate">Asssessing the SILAC isotope incorporation rate<a class="anchor" aria-label="anchor" href="#asssessing-the-silac-isotope-incorporation-rate"></a>
</h2>
<p>Before performing a typical SILAC experiment, it is necessary to confirm complete (e.g &gt; 95%) incorporation of the supplemented isotopically labelled amino acids into the cellular proteome.</p>
<p>We do this with <code><a href="https://rdrr.io/pkg/camprotR/man/estimate_incorporation.html" class="external-link">camprotR::estimate_incorporation()</a></code> function below, which wraps multiple <code>camprotR</code> functions to parse and filter the input data and then estimate the incorporation at peptide and protein level.</p>
<p>Calculating the incorporation for a given peptide is simple, e.g H/(H+L) for a typical two label experiment. However, for very highly incorporated samples, the true intensity of the L peptide ions will be close to zero, which can lead to erroneous assignment of other ions to the L peptide. <code><a href="https://rdrr.io/pkg/camprotR/man/estimate_incorporation.html" class="external-link">estimate_incorporation()</a></code> also supports the calculation of incorporation from defined mixes of heavy and light labelled material, which makes the estimation of incorporation rate more robust, since both isotopes should exist for all peptides. Here, heavy and light labelled cells were mixed 1:1.</p>
<p>Importantly, note that there is a good correlation between the light and heavy intensities, regardless whether either or both isotopes were from a spectrum match, or one peptide isotope was identified based on the expected ‘mass shift’ relative to the peptide-spectrum match (PSM) for the cognate isotope. This indicates that identified isotope pairs are correct and the the quantification values are trustable.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The cRAP fasta file for this experiment is part of this package.</span>
<span class="co"># For a real experiment, make sure to the use the same cRAP fasta that was used in PD</span>
<span class="va">crap_fasta_inf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>
  <span class="st">"extdata"</span>, <span class="st">"cRAP_20190401.fasta.gz"</span>, 
  package <span class="op">=</span> <span class="st">"Proteomics.data.analysis"</span>
<span class="op">)</span>

<span class="co"># psm_silac_p4 and pep_silac_p4 are loaded from the `camprotR` package </span>
<span class="va">incorporation_results</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/estimate_incorporation.html" class="external-link">estimate_incorporation</a></span><span class="op">(</span>
  <span class="co"># can replace with PD output using read.delim(PSMs.txt)</span>
  psm_input <span class="op">=</span> <span class="va">psm_silac_p4</span>, 
  <span class="co"># can replace with PD output using read.delim(PeptideGroups.txt)</span>
  peptide_input <span class="op">=</span> <span class="va">pep_silac_p4</span>, 
  crap_fasta <span class="op">=</span> <span class="va">crap_fasta_inf</span>,
  mix <span class="op">=</span> <span class="fl">1</span> <span class="co"># This incorporation test was performed with a 1:1 H/L mix</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">incorporation_results</span><span class="op">$</span><span class="va">HL_correlation</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-3-1.png" width="576"></p>
<p>The median observed incorporation for the mixed sample is 50.8 at the peptide-level and 50.5 at the protein level. Correcting for the mixing proportions this gives us incorporation rate estimates just over 100%, which suggests the true incorporation rate is likely to be very close to 100%. We are only using a toy data set here with 100 proteins. With the full data, we would get a more accurate estimate.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">incorporation_results</span><span class="op">$</span><span class="va">peptide_incorporation</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-4-1.png" width="384"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">incorporation_results</span><span class="op">$</span><span class="va">protein_incorporation</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-4-2.png" width="384"></p>
</div>
<div class="section level2">
<h2 id="processing-and-qc-of-silac-experimental-data">Processing and QC of SILAC experimental data<a class="anchor" aria-label="anchor" href="#processing-and-qc-of-silac-experimental-data"></a>
</h2>
<p>Now, onto some experimental data. Here, we process data from an OOPS <span class="citation">(Queiroz et al. 2019)</span> experiment designed to identify proteins that are significantly enriched upon UV crosslinking (CL) vs non-crosslinked control cells (NC). RNA-binding proteins (RBPs) should be much more abundant in CL samples, since this covalently crosslinks the RBPs to RNA, and retains them in the OOPS interface, from which the sample is taken.</p>
<p>In total, we have 4 SILAC runs, representing 4 separate plates of U-2 OS cells. Two plates (replicate 1 &amp; 2) were Heavy for CL and two plates (replicates 3 &amp; 4) were Light for CL. This is called a ‘label’ swap/switch and helps to ensure the results are independent of the labelling scheme.</p>
<p>The first step is to read in the data. As part of this, we will remove contaminant proteins and features without quantification data. These were defined using the cRAP database. Below, we parse the cRAP FASTA to extract the IDs for the cRAP proteins, in both ‘cRAP’ format and Uniprot IDs for these proteins.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load the cRAP FASTA used for the PD search</span>
<span class="va">crap_fasta</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">fasta.index</a></span><span class="op">(</span><span class="va">crap_fasta_inf</span>, seqtype <span class="op">=</span> <span class="st">"AA"</span><span class="op">)</span>


<span class="co"># Extract the non-cRAP UniProt accessions associated with each cRAP protein</span>
<span class="va">crap_accessions</span> <span class="op">&lt;-</span> <span class="va">crap_fasta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">desc</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_extract.html" class="external-link">str_extract_all</a></span><span class="op">(</span>pattern<span class="op">=</span><span class="st">"(?&lt;=\\|).*?(?=\\|)"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>We can then supply these cRAP protein IDs to <code><a href="https://rdrr.io/pkg/camprotR/man/parse_features.html" class="external-link">camprotR::parse_features</a></code> which will parse the PD output and remove features which may originate from contaminants, as well as features which don’t have a unique master protein. See <code><a href="https://rdrr.io/pkg/camprotR/man/parse_features.html" class="external-link">?parse_features</a></code> for further details, including the removal of ‘associated cRAP’. <code>parse_features</code> will output messages about the number of features at each stage in the filtering.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># These files are part this package</span>
<span class="va">replicates</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span>
<span class="va">pep_infiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">'OOPS_SILAC'</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'OOPS_'</span>, <span class="va">replicates</span>, <span class="st">'_PeptideGroups.txt'</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pep_infiles</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">replicates</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">pep_infiles</span><span class="op">)</span>
<span class="co">#&gt;                                     1                                     2 </span>
<span class="co">#&gt; "OOPS_SILAC/OOPS_1_PeptideGroups.txt" "OOPS_SILAC/OOPS_2_PeptideGroups.txt" </span>
<span class="co">#&gt;                                     3                                     4 </span>
<span class="co">#&gt; "OOPS_SILAC/OOPS_3_PeptideGroups.txt" "OOPS_SILAC/OOPS_4_PeptideGroups.txt"</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We use lapply below to run the same function on each element of the list pep_data</span>
<span class="co"># This is easier than running a for loop since the output is a new named list </span>
<span class="va">pep_data_parsed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
  <span class="va">pep_infiles</span>, <span class="kw">function</span><span class="op">(</span><span class="va">infile</span><span class="op">)</span> <span class="op">{</span> <span class="co"># define the function to run</span>
    
    <span class="va">infdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span>
      <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">infile</span>, package <span class="op">=</span> <span class="st">"Proteomics.data.analysis"</span><span class="op">)</span><span class="op">)</span>
    
    <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/parse_features.html" class="external-link">parse_features</a></span><span class="op">(</span><span class="va">infdata</span>,
                   silac <span class="op">=</span> <span class="cn">TRUE</span>,
                   level <span class="op">=</span> <span class="st">'peptide'</span>,
                   crap_proteins <span class="op">=</span> <span class="va">crap_accessions</span>,
                   unique_master <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
  <span class="op">}</span>
<span class="op">)</span></code></pre></div>
<p>Next, we add new columns to describe the intensities with respect to the CL and NC conditions, based on how H &amp; L map to CL &amp; NC for each replicate.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Define a function name to take the replicate number and annotate the data</span>

<span class="va">annotate_parsed_data</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">rep_n</span><span class="op">)</span><span class="op">{</span>
  
  <span class="va">pep_data</span> <span class="op">&lt;-</span> <span class="va">pep_data_parsed</span><span class="op">[[</span><span class="va">rep_n</span><span class="op">]</span><span class="op">]</span>
  
  <span class="co"># Identify the column names for the light and heavy intensities</span>
  <span class="co"># (these names are not consistent between samples, yours may be different!)</span>
  <span class="va">abundance_light_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">'Abundance.*.Light.Sample'</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pep_data</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="va">abundance_heavy_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">'Abundance.*.Heavy.Sample'</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pep_data</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  
  <span class="co"># Label-swap info defines whether heavy or light is CL</span>
  <span class="va">cl_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">rep_n</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="va">abundance_heavy_col</span>, <span class="va">abundance_light_col</span><span class="op">)</span> 
  <span class="va">nc_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">rep_n</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="va">abundance_light_col</span>, <span class="va">abundance_heavy_col</span><span class="op">)</span>
  
  <span class="va">pep_data</span> <span class="op">&lt;-</span> <span class="va">pep_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="co"># Add new columns with CL or NC intensities </span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="st">'CL'</span> <span class="op">=</span> <span class="op">!</span><span class="op">!</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/tidyeval.html" class="external-link">sym</a></span><span class="op">(</span><span class="va">cl_col</span><span class="op">)</span>, <span class="co"># !!sym(var) allows us to supply a string var for tidy evalulation</span>
           <span class="st">'NC'</span> <span class="op">=</span> <span class="op">!</span><span class="op">!</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/tidyeval.html" class="external-link">sym</a></span><span class="op">(</span><span class="va">nc_col</span><span class="op">)</span>,
           Replicate <span class="op">=</span> <span class="va">rep_n</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="co"># And subset to these columns</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span>,
           <span class="va">Sequence</span>,
           <span class="va">Modifications</span>,
           <span class="va">CL</span>,
           <span class="va">NC</span>,
           <span class="va">Replicate</span><span class="op">)</span>
  
  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">pep_data</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># Apply the function to all peptide data.frames</span>
<span class="va">annot_pep_data_parsed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pep_data_parsed</span><span class="op">)</span>, <span class="co"># annotate_parsed_data function using the name</span>
                          <span class="va">annotate_parsed_data</span><span class="op">)</span>

<span class="co"># Re-annotate with the names</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">annot_pep_data_parsed</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pep_data_parsed</span><span class="op">)</span></code></pre></div>
<p>Now that we have added the replicate number as a column, we can bind together the rows from all replicates to keep all our data in a single <code>data.frame</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">abundance_data_complete</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="st">'rbind'</span>, <span class="va">annot_pep_data_parsed</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">remove_rownames</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Next, we calculate the CL/NC ratio using <code><a href="https://rdrr.io/pkg/camprotR/man/get_ratio.html" class="external-link">camprotR::get_ratio</a></code>, which will also add a <code>missing</code> column to describe whether one of the quantification values is missing. <code><a href="https://rdrr.io/pkg/camprotR/man/get_ratio.html" class="external-link">camprotR::get_ratio</a></code> expects the quantification values to be on a log scale.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ratios</span> <span class="op">&lt;-</span> <span class="va">abundance_data_complete</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">CL</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">NC</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># Retain peptides where either CL and/or NC is finite</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>CL <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">CL</span><span class="op">)</span>, NC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">NC</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># log2-transform quantification</span>
  <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_ratio.html" class="external-link">get_ratio</a></span><span class="op">(</span><span class="va">CL</span>, <span class="va">NC</span>, bind <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># Obtain CL/NC ratio</span></code></pre></div>
<p>Below, we tally and plot the missing values per replicate for each method. Note that ~40% of peptides have a missing quantification value and it’s much more likely to be missing in NC than CL. This is what we expect since the OOPS method is designed to capture RBPs that are UV crosslinked to RNA, so without crosslinking, there should be very little protein present in the interface.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Tally the missing values</span>
<span class="va">missing_tallies</span> <span class="op">&lt;-</span> <span class="va">ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">missing</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">tally</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">missing_tallies</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">missing</span>, y <span class="op">=</span> <span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">'identity'</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y<span class="op">=</span><span class="st">'Peptides'</span>, x<span class="op">=</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle<span class="op">=</span><span class="fl">45</span>, vjust<span class="op">=</span><span class="fl">1</span>, hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-11-1.png" width="576"></p>
<blockquote>
<h3 id="exercise-1">Exercise 1</h3>
<p>Modify the code above so that rather than showing one bar each for every level of missing status, it shows a stacked bar for each replicate, where the fill denotes the missing status (see below for how it should look)</p>
</blockquote>
<div class="section level3">
<h3 id="solution">Solution<a class="anchor" aria-label="anchor" href="#solution"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">missing_tallies</span> <span class="op">&lt;-</span> <span class="va">ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Replicate</span>, <span class="va">missing</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">tally</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">missing_tallies</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Replicate</span>, y <span class="op">=</span> <span class="va">n</span>, fill <span class="op">=</span> <span class="va">missing</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">'identity'</span>, position <span class="op">=</span> <span class="st">'fill'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">'Fraction'</span>, fill<span class="op">=</span><span class="st">''</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-12-1.png" width="576"></p>
<p>Given the large number of missing values, there is a concern that protein abundance in NC may be very low. Therefore, NC peptides identified by mass shift rather than peptide spectrum matching could be incorrect. Note that peptide sequences identified by PSM are referred to as having been ‘matched’ in some code blocks below.</p>
<p>To get the information about what quantification values come from PSMs, we need to interrogate the PSM-level output.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># These files are part of this package</span>
<span class="va">psm_infiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">'_PeptideGroups.txt'</span>, <span class="st">'_PSMs.txt'</span>, <span class="va">pep_infiles</span><span class="op">)</span></code></pre></div>
<p>Below we use <code><a href="https://rdrr.io/pkg/camprotR/man/silac_psm_seq_int.html" class="external-link">camprotR::silac_psm_seq_int</a></code> to identify which quantification values are from PSMs.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">psm_matched_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
  <span class="va">psm_infiles</span>, <span class="kw">function</span><span class="op">(</span><span class="va">infile</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">infdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span>
      <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">infile</span>, package <span class="op">=</span> <span class="st">"Proteomics.data.analysis"</span><span class="op">)</span><span class="op">)</span>
    
    <span class="co"># If you export from PD including the 'Sequence' column, </span>
    <span class="co"># you don't need this step.</span>
    <span class="co"># Below, we convert the Annotated.Sequence column into a Sequence column</span>
    <span class="va">infdata</span><span class="op">$</span><span class="va">Sequence</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">infdata</span><span class="op">$</span><span class="va">Annotated.Sequence</span>, split <span class="op">=</span> <span class="st">'\\.'</span><span class="op">)</span>, <span class="st">'[['</span>, <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
      <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">toupper</a></span><span class="op">(</span><span class="op">)</span>
    
    <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/silac_psm_seq_int.html" class="external-link">silac_psm_seq_int</a></span><span class="op">(</span><span class="va">infdata</span>, sequence_col <span class="op">=</span> <span class="st">'Sequence'</span><span class="op">)</span>
  <span class="op">}</span>
<span class="op">)</span></code></pre></div>
<p>Then we bind together the spectrum matched data into one data frame, ready to merge with the peptide quantification.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_psm_matched_data</span> <span class="op">&lt;-</span> <span class="va">psm_matched_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">rep_n</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">psm_matched_data</span><span class="op">[[</span><span class="va">rep_n</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Replicate <span class="op">=</span> <span class="va">rep_n</span><span class="op">)</span>
  <span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Below, we merge the quantification and PSM information and make new columns to describe whether the CL/NC quantification is from a PSM. Note that we need to update the <code>Modifications</code> and <code>Sequence</code> columns in the peptide-level output so they can be matched to the columns in the PSM-level columns.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># merge the matched information and add new columns for the CL/NC matched information</span>
<span class="va">ratios_matched</span> <span class="op">&lt;-</span> <span class="va">ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  
  <span class="co"># Update the modifications column for the peptide object so it </span>
  <span class="co"># doesn't include SILAC modifications </span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Modifications <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/remove_silac_modifications.html" class="external-link">remove_silac_modifications</a></span><span class="op">(</span><span class="va">Modifications</span>, level <span class="op">=</span> <span class="st">'peptide'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  
  <span class="co"># Update the sequence column to all uppercase</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rowwise.html" class="external-link">rowwise</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Sequence <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">toupper</a></span><span class="op">(</span><span class="va">Sequence</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  
  <span class="co"># Merge with the sequenced information</span>
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">all_psm_matched_data</span>, 
        by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Sequence'</span>, <span class="st">'Modifications'</span>, <span class="st">'Replicate'</span><span class="op">)</span><span class="op">)</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  
  <span class="co"># Add new columns with CL/NC matched information using the</span>
  <span class="co"># matched_Heavy and matched_Light columns and the label swap information</span>
  <span class="co"># (rep1/2, CL=H; rep3/4 CL=L)</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="st">'Matched_CL'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Replicate</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="va">matched_Heavy</span>, <span class="va">matched_Light</span><span class="op">)</span>,
         <span class="st">'Matched_NC'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Replicate</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="va">matched_Light</span>, <span class="va">matched_Heavy</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  
  <span class="co"># And subset to these columns</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span>,
         <span class="va">Sequence</span>, 
         <span class="va">Modifications</span>,
         <span class="va">Replicate</span>,
         <span class="va">ratio</span>,
         <span class="va">CL</span>, 
         <span class="va">NC</span>,
         <span class="va">Matched_CL</span>,
         <span class="va">Matched_NC</span>,
         <span class="va">missing</span><span class="op">)</span>

<span class="co"># Add a single column to describe the matched information across the two conditions</span>
<span class="va">ratios_matched</span> <span class="op">&lt;-</span> <span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>
    matched <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interaction.html" class="external-link">interaction</a></span><span class="op">(</span><span class="va">Matched_CL</span>, <span class="va">Matched_NC</span><span class="op">)</span>,
    matched <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/recode.html" class="external-link">recode</a></span><span class="op">(</span><span class="va">matched</span>,
                            <span class="st">'TRUE.TRUE'</span><span class="op">=</span><span class="st">'Both spectrum matched'</span>,
                            <span class="st">'TRUE.FALSE'</span><span class="op">=</span><span class="st">'CL spectrum matched'</span>,
                            <span class="st">'FALSE.TRUE'</span><span class="op">=</span><span class="st">'NC spectrum matched'</span><span class="op">)</span>, 
                     levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Both spectrum matched'</span>, 
                                <span class="st">'CL spectrum matched'</span>, 
                                <span class="st">'NC spectrum matched'</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<p>Below, we consider how often the ratio comes from a peptide where both CL and NC were spectrum matched, or just one spectrum matched, and the other one therefore being by mass shift. Note that a minority of peptides have both CL and NC sequenced and the majority are just sequenced in CL.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Tally the peptide match status</span>
<span class="va">matched_tallies</span> <span class="op">&lt;-</span> <span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">ratio</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Replicate</span>, <span class="va">matched</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">tally</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Plot as stacked bar plot</span>
<span class="va">matched_tallies</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Replicate</span>, y <span class="op">=</span> <span class="va">n</span>, fill <span class="op">=</span> <span class="va">matched</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">'identity'</span>, position<span class="op">=</span><span class="st">'fill'</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_cat_palette.html" class="external-link">get_cat_palette</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">''</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">'Replicate'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Fraction of peptides'</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-17-1.png" width="576"></p>
<p>We can then see consider how the correlated between CL and NC relates to the ‘spectrum matched’ information. Note that when both CL and NC are spectrum matched, there are two clear populations, especially for replicates 1 &amp; 2.</p>
<ol style="list-style-type: decimal">
<li>CL == NC (on grey dashed line). These are the peptides not enriched by UV CL and therefore from non-RBPs</li>
<li>CL &gt;&gt; NC (below grey dashed line). These peptides are heavily enriched by UV CL and therefore from RBPs.</li>
</ol>
<p>When just CL is matched, the correlation is less clear and the two populations are not obvious. One plausible explanation is that many of the NC quantification values obtained when CL is identified by spectrum matching, but NC is identified by mass shift, are from ions erroneously assigned to a peptide. This would be a concern.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">CL</span>, y <span class="op">=</span> <span class="va">NC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.5</span>, alpha<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, linetype <span class="op">=</span> <span class="fl">2</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span> <span class="co"># line at CL==NC</span>
  <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">15</span>, border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">Replicate</span><span class="op">~</span> <span class="va">matched</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">14</span>, <span class="fl">26</span>, <span class="fl">4</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">14</span>, <span class="fl">26</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">14</span>, <span class="fl">26</span>, <span class="fl">4</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">14</span>, <span class="fl">26</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">'CL (log2)'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'NC (log2)'</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 4762 rows containing missing values (geom_point).</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-18-1.png" width="768"></p>
<p>Below, we quantify the correlations</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cl_nc_correlations</span> <span class="op">&lt;-</span> <span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">CL</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">NC</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">matched</span>, <span class="va">Replicate</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>cor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">CL</span>, <span class="va">NC</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; `summarise()` has grouped output by 'matched'. You can override using the</span>
<span class="co">#&gt; `.groups` argument.</span>

<span class="va">cl_nc_correlations</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">matched</span>, <span class="va">cor</span>, colour<span class="op">=</span><span class="va">Replicate</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_size<span class="op">=</span><span class="fl">15</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle<span class="op">=</span><span class="fl">45</span>, vjust<span class="op">=</span><span class="fl">1</span>, hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Pearson correlation'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_cat_palette.html" class="external-link">get_cat_palette</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p>Whenever a SILAC experiment is performed where complete absence of a protein in one condition is likely, it is recommended to perform this QC step. If the correlation is very poor when just one condition is spectrum matched, it may be necessary to exclude these peptides.</p>
<p>In this vignette example, we will proceed with all the peptides, but retain information about the spectrum matching in the protein quantification, should we need to investigate this further.</p>
<p>Now, we want to make an <code>MSnSet</code> from the peptide-level CL/NC ratios. For this, we need to generate a matrix of quantification values (CL/NC ratios), where rows are features (peptides), and a separate data.frame with information about the features, e.g master protein assignment. See the <code>vignette("msnset", package="camprotR")</code> for more details about MSnSets.</p>
<p>Since each of the SILAC samples was processed separately, the assignments of peptides -&gt; master proteins may not be consistent. This will cause issues when we try and create the MSnSet and we therefore need to update the master protein column.</p>
<p><code><a href="https://rdrr.io/pkg/camprotR/man/get_parsimony_pep2prot.html" class="external-link">camprotR::get_parsimony_pep2prot()</a></code> will take a list of files, each with Sequence, Protein.Acession and Master.Protein.Accession columns and generate a unified peptide sequence to master protein assignment. This is done by an approximately parsimonous approach, which is not identical to PD. The function therefore also summarises the difference between the PD assignments and the updated assignment, which is normally very little.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">pep_infiles</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, package <span class="op">=</span> <span class="st">"Proteomics.data.analysis"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Peptide.Groups.Peptide.Group.ID"                      </span>
<span class="co">#&gt;  [2] "Checked"                                              </span>
<span class="co">#&gt;  [3] "Confidence"                                           </span>
<span class="co">#&gt;  [4] "Sequence"                                             </span>
<span class="co">#&gt;  [5] "Modifications"                                        </span>
<span class="co">#&gt;  [6] "Qvality.PEP"                                          </span>
<span class="co">#&gt;  [7] "Qvality.q.value"                                      </span>
<span class="co">#&gt;  [8] "Number.of.Protein.Groups"                             </span>
<span class="co">#&gt;  [9] "Number.of.Proteins"                                   </span>
<span class="co">#&gt; [10] "Number.of.PSMs"                                       </span>
<span class="co">#&gt; [11] "Master.Protein.Accessions"                            </span>
<span class="co">#&gt; [12] "Protein.Accessions"                                   </span>
<span class="co">#&gt; [13] "Number.of.Missed.Cleavages"                           </span>
<span class="co">#&gt; [14] "Theo.MHplus.in.Da"                                    </span>
<span class="co">#&gt; [15] "Abundance.F26.Light.Sample"                           </span>
<span class="co">#&gt; [16] "Abundance.F26.Heavy.Sample"                           </span>
<span class="co">#&gt; [17] "Precursor.Quan.Result.ID"                             </span>
<span class="co">#&gt; [18] "Quan.Info"                                            </span>
<span class="co">#&gt; [19] "Quan.Channel"                                         </span>
<span class="co">#&gt; [20] "Found.in.Sample.in.S42.F26.Heavy.Sample"              </span>
<span class="co">#&gt; [21] "Found.in.Sample.in.S26.F26.Light.Sample"              </span>
<span class="co">#&gt; [22] "Confidence.by.Search.Engine.MS.Amanda.20"             </span>
<span class="co">#&gt; [23] "Confidence.by.Search.Engine.Sequest.HT"               </span>
<span class="co">#&gt; [24] "Percolator.q.Value.by.Search.Engine.MS.Amanda.20"     </span>
<span class="co">#&gt; [25] "Percolator.q.Value.by.Search.Engine.Sequest.HT"       </span>
<span class="co">#&gt; [26] "Percolator.PEP.by.Search.Engine.MS.Amanda.20"         </span>
<span class="co">#&gt; [27] "Percolator.PEP.by.Search.Engine.Sequest.HT"           </span>
<span class="co">#&gt; [28] "Amanda.Score.by.Search.Engine.MS.Amanda.20"           </span>
<span class="co">#&gt; [29] "CharmeRT.Combined.Score.by.Search.Engine.MS.Amanda.20"</span>
<span class="co">#&gt; [30] "XCorr.by.Search.Engine.Sequest.HT"                    </span>
<span class="co">#&gt; [31] "Top.Apex.RT.in.min"</span>
<span class="co"># Create a unified peptide sequence to protein assignment</span>
<span class="va">new_seq_to_master</span> <span class="op">&lt;-</span> <span class="va">pep_infiles</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">x</span>, package <span class="op">=</span> <span class="st">"Proteomics.data.analysis"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_parsimony_pep2prot.html" class="external-link">get_parsimony_pep2prot</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; With original assignments, 1951 master proteins. With update, 1820 master proteins</span>
<span class="co">#&gt; Storing counts in `nn`, as `n` already present in input</span>
<span class="co">#&gt; Storing counts in `nn`, as `n` already present in input</span>
<span class="co">#&gt; Comparing Master Protein IDs</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `name = "new_name"` to pick a new name.</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-20-1.png" width="480"></p>
<pre><code><span class="co">#&gt; .</span>
<span class="co">#&gt; Different ID(s)      Same ID(s) </span>
<span class="co">#&gt;            1188           18372 </span>
<span class="co">#&gt; .</span>
<span class="co">#&gt; Different ID(s)      Same ID(s) </span>
<span class="co">#&gt;             6.1            93.9 </span>
<span class="co">#&gt;                   Updated:Multiple Updated:Single</span>
<span class="co">#&gt; Original:Multiple              106            629</span>
<span class="co">#&gt; Original:Single                384          18441</span>
<span class="co">#&gt;                   Updated:Multiple Updated:Single</span>
<span class="co">#&gt; Original:Multiple             0.54           3.22</span>
<span class="co">#&gt; Original:Single               1.96          94.28</span></code></pre>
<p>Next, we remove peptides without both CL and NC, e.g no quantified ratio, and then merge with the updated master protein assignments</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove peptides without both CL + NC quantified</span>
<span class="va">flt_ratios</span> <span class="op">&lt;-</span> <span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">ratio</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">new_seq_to_master</span>, by<span class="op">=</span><span class="st">'Sequence'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="co"># replace Master.Protein.Accessions with Updated.Master.Protein.Accessions</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Master.Protein.Accessions<span class="op">=</span><span class="va">Updated.Master.Protein.Accessions</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R">

<span class="co"># Create a wide table with unique ids as row names</span>
<span class="va">flt_ratios_wide</span> <span class="op">&lt;-</span> <span class="va">flt_ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span>,
         <span class="va">Sequence</span>, <span class="va">Modifications</span>,
         <span class="va">Replicate</span>, <span class="va">Matched_NC</span>, <span class="va">ratio</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="st">"Replicate"</span>, values_from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Matched_NC"</span>, <span class="st">"ratio"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="co"># Create an id column from the sequence and modification columns</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/unite.html" class="external-link">unite</a></span><span class="op">(</span><span class="va">id</span>, <span class="va">Sequence</span>, <span class="va">Modifications</span>, remove <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span>

<span class="co"># Create expression matrix (exprs)</span>
<span class="va">exprs_data</span> <span class="op">&lt;-</span> <span class="va">flt_ratios_wide</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"ratio_[1-4]"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># select unique ids and ratio data</span>
  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Create feature metadata data frame (fData)</span>
<span class="va">feat_data</span> <span class="op">&lt;-</span> <span class="va">flt_ratios_wide</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"ratio"</span><span class="op">)</span><span class="op">)</span> <span class="co"># select unique ids and everything but ratio data</span>

<span class="co"># Create MSnSet</span>
<span class="va">pep_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnSet</a></span><span class="op">(</span>exprs <span class="op">=</span> <span class="va">exprs_data</span>,
                  fData <span class="op">=</span> <span class="va">feat_data</span><span class="op">)</span>

<span class="co"># Add replicate number to phenotype data</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">pep_res</span><span class="op">)</span><span class="op">$</span><span class="va">Replicate</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">exprs_data</span><span class="op">)</span>

<span class="va">pep_res</span>
<span class="co">#&gt; MSnSet (storageMode: lockedEnvironment)</span>
<span class="co">#&gt; assayData: 3839 features, 4 samples </span>
<span class="co">#&gt;   element names: exprs </span>
<span class="co">#&gt; protocolData: none</span>
<span class="co">#&gt; phenoData</span>
<span class="co">#&gt;   sampleNames: ratio_1 ratio_2 ratio_4 ratio_3</span>
<span class="co">#&gt;   varLabels: Replicate</span>
<span class="co">#&gt;   varMetadata: labelDescription</span>
<span class="co">#&gt; featureData</span>
<span class="co">#&gt;   featureNames: AAAAAAAAAAAAAAAGAGAGAK_ AAAAAAAAAAGAAGGR_1xAcetyl</span>
<span class="co">#&gt;     [N-Term] ... YYVTIIDAPGHR_ (3839 total)</span>
<span class="co">#&gt;   fvarLabels: Master.Protein.Accessions Sequence ... Matched_NC_3 (7</span>
<span class="co">#&gt;     total)</span>
<span class="co">#&gt;   fvarMetadata: labelDescription</span>
<span class="co">#&gt; experimentData: use 'experimentData(object)'</span>
<span class="co">#&gt; Annotation:  </span>
<span class="co">#&gt; - - - Processing information - - -</span>
<span class="co">#&gt;  MSnbase version: 2.22.0</span></code></pre></div>
<p>Then we summarise the peptide-level ratios to protein-level ratios. Here, we’ll use the median.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prot_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/combineFeatures.html" class="external-link">combineFeatures</a></span><span class="op">(</span>
  <span class="va">pep_res</span>, 
  groupBy <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_res</span><span class="op">)</span><span class="op">$</span><span class="va">Master.Protein.Accessions</span>,
  method <span class="op">=</span> <span class="st">"median"</span>,
  na.rm <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span> 
<span class="co">#&gt; Your data contains missing values. Please read the relevant section in</span>
<span class="co">#&gt; the combineFeatures manual page for details on the effects of missing</span>
<span class="co">#&gt; values on data aggregation.</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pep_res</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 3839</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 1128</span></code></pre></div>
<p>Note that the summarisation is occurring on the expression data, but will affect the feature data too (the values in the first feature are taken). If we want to retain information about whether the peptide was ‘matched’ in both conditions, we will need to generate this ourselves and re-attach to the MSnSet.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prot_matched_nc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_res</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">Sequence</span>, <span class="op">-</span><span class="va">Modifications</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="va">Master.Protein.Accessions</span>, values_to <span class="op">=</span> <span class="st">'Matched_NC'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Matched_NC</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>any_not_sequenced <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="op">!</span><span class="va">Matched_NC</span><span class="op">)</span><span class="op">)</span>


<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">prot_matched_nc</span>, by <span class="op">=</span> <span class="st">"Master.Protein.Accessions"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span></code></pre></div>
<p>We can now inspect the completeness of our protein-level data using <code><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">MSnbase::plotNA()</a></code>. In this case 383/1128 proteins have complete data (all 4 replicates)</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">plotNA</a></span><span class="op">(</span><span class="va">prot_res</span>, pNA <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-25-1.png" width="384"></p>
<blockquote>
<h3 id="exercise-2">Exercise 2</h3>
<p>We have used <code><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">MSnbase::plotNA</a></code> to assess the missing values but it’s straightforward to do this ourselves directly from the <code>prot_res</code> object. Remembering that <code><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs()</a></code> can be used to extract the abundance matrix from an <code>MSnSet</code>, answer the following:</p>
<ol style="list-style-type: decimal">
<li>How many values are missing in total?</li>
<li>What fraction of values are missing?</li>
<li>How many missing values are there in each sample?</li>
<li>How many proteins have exactly 1 missing value?</li>
</ol>
</blockquote>
</div>
<div class="section level3">
<h3 id="solution-1">Solution<a class="anchor" aria-label="anchor" href="#solution-1"></a>
</h3>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span><span class="op">)</span> <span class="co">#1</span>
<span class="co">#&gt; [1] 1732</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span><span class="op">)</span> <span class="co">#2</span>
<span class="co">#&gt; [1] 0.3838652</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span><span class="op">)</span> <span class="co">#3 </span>
<span class="co">#&gt; ratio_1 ratio_2 ratio_4 ratio_3 </span>
<span class="co">#&gt;     466     472     364     430</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co">#4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   0   1   2   3 </span>
<span class="co">#&gt; 383 154 195 396</span></code></pre></div>
<p>To identify proteins with significantly increased abundance in CL vs NC, we will use DEqMS <span class="citation">(Zhu et al. 2020)</span>, which you can think of as an extension of limma <span class="citation">(Ritchie et al. 2015)</span> specifically for proteomics.</p>
<p>The next step is taken from the <a href="https://bioconductor.org/packages/release/bioc/vignettes/DEqMS/inst/doc/DEqMS-package-vignette.html#extract-quant-data-columns-for-deqms" class="external-link">DEqMS vignette</a>. The idea is that we want to create an <code>MArrayLM</code> object as per normal <code>limma</code> analysis and then add a <code>$count</code> column to the <code>MArrayLM</code> object and then use the <code>spectraCounteBayes</code> function to perform the Bayesian shrinkage using the count column (which describes the number of pepitdes per protein) rather than the <code>$Amean</code> column (which describes the mean CL:NC ratio). In the DEqMS paper, they suggest that the best summarisation metric to use is the minimum value across the samples, so our <code>count</code> columns is the minimum number of peptides per protein.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="va">prot_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">filterNA</a></span><span class="op">(</span>pNA <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># select proteins in min. 2/4 reps</span>
  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>Intercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">dat</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">efit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>

<span class="va">protein_ratio_long</span> <span class="op">&lt;-</span> <span class="va">prot_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">'Master.Protein.Accessions'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols<span class="op">=</span><span class="op">-</span><span class="va">Master.Protein.Accessions</span>, values_to<span class="op">=</span><span class="st">'protein_ratio'</span>, names_to<span class="op">=</span><span class="st">'sample'</span><span class="op">)</span>

<span class="co"># Obtain the min peptide count across the samples</span>
<span class="va">min_pep_count</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/count_features_per_protein.html" class="external-link">count_features_per_protein</a></span><span class="op">(</span><span class="va">pep_res</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">protein_ratio_long</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Master.Protein.Accessions'</span>, <span class="st">'sample'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">protein_ratio</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>min_pep_count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span>

<span class="co"># add the min peptide count</span>
<span class="va">efit</span><span class="op">$</span><span class="va">count</span> <span class="op">&lt;-</span> <span class="va">min_pep_count</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">efit</span><span class="op">$</span><span class="va">coefficients</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">min_pep_count</span><span class="op">)</span>

<span class="co"># run DEqMS</span>
<span class="va">efit_deqms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/DEqMS/man/spectraCounteBayes.html" class="external-link">spectraCounteBayes</a></span><span class="op">(</span><span class="va">efit</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>In this case the relationship between peptide count and variance is only really apparent when the minimum number of peptides from which a protein-level ratio is obtained is very low. Typically, one might remove quantifications where there is just a single peptide for a protein. Here, we will keep them, and we can refer to the <code>count</code> column in the final results object if we want to check the minimum numbber of peptides observed per sample.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Diagnostic plots</span>
<span class="fu"><a href="https://rdrr.io/pkg/DEqMS/man/VarianceBoxplot.html" class="external-link">VarianceBoxplot</a></span><span class="op">(</span><span class="va">efit_deqms</span>, n <span class="op">=</span> <span class="fl">30</span>, xlab <span class="op">=</span> <span class="st">"Peptides"</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-28-1.png" width="384"></p>
<p>Below, we summarise the number of proteins with statistically different abundance in CL vs NC and plot a ‘volcano’ plot to visualise this. Note that all proteins with a statistically significant change are increased in CL, so the plot looks more like a fire-hose than a volcano!</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">deqms_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DEqMS/man/outputResult.html" class="external-link">outputResult</a></span><span class="op">(</span><span class="va">efit_deqms</span>, coef_col <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">deqms_results</span><span class="op">$</span><span class="va">sca.adj.pval</span> <span class="op">&lt;</span> <span class="fl">0.01</span>, <span class="st">'sig.'</span>, <span class="st">'not sig.'</span><span class="op">)</span>,
      <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">deqms_results</span><span class="op">$</span><span class="va">logFC</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">'Higher in CL'</span>, <span class="st">'Lower in CL'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;           </span>
<span class="co">#&gt;            Higher in CL Lower in CL</span>
<span class="co">#&gt;   not sig.          409          77</span>
<span class="co">#&gt;   sig.              246           0</span>

<span class="va">deqms_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">logFC</span>, y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">sca.P.Value</span><span class="op">)</span>, colour <span class="op">=</span> <span class="va">sca.adj.pval</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/theme_camprot.html" class="external-link">theme_camprot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'grey'</span>, <span class="fu"><a href="https://rdrr.io/pkg/camprotR/man/get_cat_palette.html" class="external-link">get_cat_palette</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">'CL vs NC Sig.'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'CL vs NC (Log2)'</span>, y <span class="op">=</span> <span class="st">'-log10(p-value)'</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-29-1.png" width="480"></p>
<p>Finally, we can explore and, if we wish, export our results. Importantly, the <code>$t</code>, <code>$P.Value</code> and <code>$adj.P.Val</code> columnts are from <code>limma</code>. The columns prefixed with <code>sca</code> are the from <code>DEqMS</code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">deqms_results</span><span class="op">)</span>
<span class="co">#&gt;           logFC  AveExpr        t      P.Value    adj.P.Val        B   gene</span>
<span class="co">#&gt; P15880 5.843093 5.843093 28.13353 2.300892e-07 0.0001590016 7.528938 P15880</span>
<span class="co">#&gt; P22626 6.561859 6.561859 25.16140 4.344306e-07 0.0001590016 7.084582 P22626</span>
<span class="co">#&gt; Q13148 4.920545 4.920545 22.69799 7.802209e-07 0.0001607499 6.640967 Q13148</span>
<span class="co">#&gt; P43243 4.424089 4.424089 18.70247 2.336893e-06 0.0001607499 5.728685 P43243</span>
<span class="co">#&gt; P23588 5.250793 5.250793 20.13712 1.538278e-06 0.0001607499 6.088209 P23588</span>
<span class="co">#&gt; P84103 5.857123 5.857123 20.66152 1.329806e-06 0.0001607499 6.210104 P84103</span>
<span class="co">#&gt;        count    sca.t  sca.P.Value sca.adj.pval</span>
<span class="co">#&gt; P15880     4 35.89893 4.910590e-08 3.594552e-05</span>
<span class="co">#&gt; P22626     7 29.58960 1.498581e-07 5.484808e-05</span>
<span class="co">#&gt; Q13148     1 24.37400 4.578182e-07 7.341608e-05</span>
<span class="co">#&gt; P43243     7 22.91838 6.522450e-07 7.341608e-05</span>
<span class="co">#&gt; P23588     3 22.86445 6.611324e-07 7.341608e-05</span>
<span class="co">#&gt; P84103     1 21.57023 9.237939e-07 7.341608e-05</span></code></pre></div>
<div id="refs" class="references">
<div id="ref-http://zotero.org/users/5634351/items/6HPNEJ3F">
<p>Krüger, Marcus, Markus Moser, Siegfried Ussar, Ingo Thievessen, Christian A. Luber, Francesca Forner, Sarah Schmidt, Sara Zanivan, Reinhard Fässler, and Matthias Mann. 2008. “SILAC Mouse for Quantitative Proteomics Uncovers Kindlin-3 as an Essential Factor for Red Blood Cell Function.” <em>Cell</em> 134 (2): 353–64. <a href="https://doi.org/10.1016/j.cell.2008.05.033" class="external-link">https://doi.org/10.1016/j.cell.2008.05.033</a>.</p>
</div>
<div id="ref-http://zotero.org/users/5634351/items/ST368TPT">
<p>Ong, Shao-En, Blagoy Blagoev, Irina Kratchmarova, Dan Bach Kristensen, Hanno Steen, Akhilesh Pandey, and Matthias Mann. 2002. “Stable Isotope Labeling by Amino Acids in Cell Culture, SILAC, as a Simple and Accurate Approach to Expression Proteomics*.” <em>Molecular &amp; Cellular Proteomics</em> 1 (5): 376–86. <a href="https://doi.org/10.1074/mcp.M200025-MCP200" class="external-link">https://doi.org/10.1074/mcp.M200025-MCP200</a>.</p>
</div>
<div id="ref-http://zotero.org/users/5634351/items/EJHWF46N">
<p>Queiroz, Rayner M. L., Tom Smith, Eneko Villanueva, Maria Marti-Solano, Mie Monti, Mariavittoria Pizzinga, Dan-Mircea Mirea, et al. 2019. “Comprehensive Identification of RNA–Protein Interactions in Any Organism Using Orthogonal Organic Phase Separation (OOPS).” <em>Nature Biotechnology</em> 37 (2): 169. <a href="https://doi.org/10.1038/s41587-018-0001-2" class="external-link">https://doi.org/10.1038/s41587-018-0001-2</a>.</p>
</div>
<div id="ref-http://zotero.org/users/5634351/items/6KTXTWME">
<p>Ritchie, Matthew E., Belinda Phipson, Di Wu, Yifang Hu, Charity W. Law, Wei Shi, and Gordon K. Smyth. 2015. “Limma Powers Differential Expression Analyses for RNA-Sequencing and Microarray Studies.” <em>Nucleic Acids Research</em> 43 (7): e47. <a href="https://doi.org/10.1093/nar/gkv007" class="external-link">https://doi.org/10.1093/nar/gkv007</a>.</p>
</div>
<div id="ref-http://zotero.org/users/5634351/items/RTM6NFVU">
<p>Zhu, Yafeng, Lukas M. Orre, Yan Zhou Tran, Georgios Mermelekas, Henrik J. Johansson, Alina Malyutina, Simon Anders, and Janne Lehtiö. 2020. “DEqMS: A Method for Accurate Variance Estimation in Differential Protein Expression Analysis.” <em>Molecular &amp; Cellular Proteomics : MCP</em> 19 (6): 1047–57. <a href="https://doi.org/10.1074/mcp.TIR119.001646" class="external-link">https://doi.org/10.1074/mcp.TIR119.001646</a>.</p>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tom Smith.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
